src/univariate/continuous/generalizedpareto.jl

```diff
@@ -169,6 +169,189 @@ function quantile(d::GeneralizedPareto{T}, p::Real) where T<:Real
+    posterior mean of ``\\theta = \\frac{\\xi}{\\sigma}``.
+
+[^ZhangStephens2009]: Jin Zhang & Michael A. Stephens (2009)
+                      A New and Efficient Estimation Method for the Generalized Pareto Distribution,
+                      Technometrics, 51:3, 316-325,
+                      DOI: [10.1198/tech.2009.08017](https://doi.org/10.1198/tech.2009.08017)
+[^Zhang2010]: Jin Zhang (2010) Improving on Estimation for the Generalized Pareto Distribution,
+              Technometrics, 52:3, 335-339,
+              DOI: [10.1198/TECH.2010.09206](https://doi.org/10.1198/TECH.2010.09206)
+"""
+function StatsBase.fit(::Type{<:GeneralizedPareto}, x::AbstractArray; μ::Real, kwargs...)
+    return fit(GeneralizedParetoKnownMu(μ), x; kwargs...)
+end
+function StatsBase.fit(g::GeneralizedParetoKnownMu, x::AbstractArray; kwargs...)
+    return fit_empiricalbayes(g, x; kwargs...)
+end
+
+# Note: our ξ is ZhangStephens2009's -k, and our θ is ZhangStephens2009's -θ
+
+function fit_empiricalbayes(
+    g::GeneralizedParetoKnownMu,
+    x::AbstractArray;
+    sorted::Bool=issorted(vec(x)),
+    improved::Bool=true,
+    min_points::Int=30,
+)
+    μ = g.μ
+    T = Base.promote_eltype(x, μ)
+    # fitting is faster when the data are sorted
+    xsorted = sorted ? vec(x) : sort(vec(x))
+    xmin, xmax = @inbounds xsorted[1], xsorted[end]
+    if xmin ≈ xmax
+        # support is nearly a point. solution is not unique; any solution satisfying the
+        # constraints σ/ξ ≈ 0 and ξ < 0 is acceptable. we choose the ξ = -1 solution, i.e.
+        # the uniform distribution
+        return GeneralizedPareto(μ, max(eps(zero(T)), xmax - μ), -one(μ))
+    end
+    # estimate θ using empirical bayes
+    θ_hat = _fit_gpd_θ_empirical_bayes(μ, xsorted, min_points, improved)
+    # estimate remaining parameters using MLE
+    return fit_mle(GeneralizedPareto, xsorted; μ=μ, θ=θ_hat)
+end
+
+# estimate θ̂ = ∫θp(θ|x,μ)dθ, i.e. the posterior mean using quadrature over grid
+# of minimum length `min_points + floor(sqrt(length(x)))` uniformly sampled over an
+# empirical prior
+function _fit_gpd_θ_empirical_bayes(μ, xsorted, min_points, improved)
+    T = Base.promote_eltype(xsorted, μ)
+    n = length(xsorted)
+
+    # empirical prior on y = r/(xmax-μ) + θ
+    θ_prior = if improved
+        _gpd_empirical_prior_improved(μ, xsorted, n)
+    else
+        _gpd_empirical_prior(μ, xsorted, n)
+    end
+
+    # quadrature points uniformly spaced on the quantiles of the θ prior
+    npoints = min_points + floor(Int, sqrt(n))
+    p = ((1:npoints) .- one(T)/2) ./ npoints
+    θ = quantile.(Ref(θ_prior), p)
+
+    # estimate mean θ over the quadrature points
+    # with weights as the normalized profile likelihood 
+    lθ = _gpd_profile_loglikelihood.(μ, θ, Ref(xsorted), n)
+    lθ_norm = logsumexp(lθ)
+    θ_hat = @inbounds sum(1:npoints) do j
+        wⱼ = exp(lθ[j] - lθ_norm)
+        return θ[j] * wⱼ
+    end
```

Comment:
It seems you could just use

```suggestion
    softmax!(lθ)
    θ_hat = dot(lθ, θ)
```